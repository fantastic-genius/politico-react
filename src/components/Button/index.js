import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ id, type, className, value }) => (
    <button type={ type } id={ id } className={ className }>{ value }</button>
);

Button.propTypes = {
  id: PropTypes.string,
  type: PropTypes.string,
  className: PropTypes.string,
  value: PropTypes.string,

}

Button.defaultTypes = {
  id: null,
  type: null,
  className: null,
  value: null
}

export default Button;
