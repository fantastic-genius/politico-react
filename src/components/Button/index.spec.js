import React from 'react';
import { shallow } from 'enzyme';
import Button from './index';

describe('<Button />', () => {
  describe('Rendering', () => {
    it('should render Footer correctly in debug mode', () => {
        const component = shallow(<Button debug/>);
        expect(component).toMatchSnapshot();
    });
  })

})