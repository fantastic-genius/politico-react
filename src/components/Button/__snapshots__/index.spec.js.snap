// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Rendering should render Footer correctly in debug mode 1`] = `<button />`;
